language/move-lang/src/parser/syntax.rs

```diff
@@ -392,6 +392,78 @@ fn parse_visibility(tokens: &mut Lexer) -> Result<Visibility, Error> {
         }
     })
 }
+// Parse an attribute value. Either a value literal or a module access
+//      AttributeValue =
+//          <Value>
+//          | <ModuleAccess>
+fn parse_attribute_value(tokens: &mut Lexer) -> Result<AttributeValue, Error> {
+    if tokens.peek() == Tok::NumValue {
+        let loc = current_token_loc(tokens);
+        return Ok(sp(loc, AttributeValue_::NumValue(parse_num(tokens)?)));
+    }
+
+    if let Some(v) = maybe_parse_value(tokens)? {
+        return Ok(sp(v.loc, AttributeValue_::Value(v)));
+    }
+
+    let ma = parse_module_access(tokens, || "attribute name value".to_string())?;
+    Ok(sp(ma.loc, AttributeValue_::ModuleAccess(ma)))
+}
+
+// Parse a single attribute
+//      Attribute =
+//          <Identifier>
+//          | <Identifier> "=" <AttributeValue>
+//          | <Identifier> "(" Comma<Attribute> ")"
+fn parse_attribute(tokens: &mut Lexer) -> Result<Attribute, Error> {
+    let start_loc = tokens.start_loc();
+    let n = parse_identifier(tokens)?;
+    let attr_ = match tokens.peek() {
+        Tok::Equal => {
+            tokens.advance()?;
+            Attribute_::Assigned(n, parse_attribute_value(tokens)?)
+        }
+        Tok::LParen => {
+            let start_loc = tokens.start_loc();
+            tokens.advance()?;
+
+            let args_ = parse_comma_list_after_start(
+                tokens,
+                start_loc,
+                Tok::LParen,
+                Tok::RParen,
+                parse_attribute,
+                "attribute",
+            )?;
+            let end_loc = tokens.previous_end_loc();
+            Attribute_::Parameterized(n, spanned(tokens.file_name(), start_loc, end_loc, args_))
+        }
+        _ => Attribute_::Name(n),
+    };
+    let end_loc = tokens.previous_end_loc();
+    Ok(spanned(tokens.file_name(), start_loc, end_loc, attr_))
+}
+
+// Parse attributes. Used to annotate a variety of AST nodes
+//      Attributes = ("#[" Comma<Attribute> "]")*
+fn maybe_parse_attributes(tokens: &mut Lexer) -> Result<Vec<Attributes>, Error> {
+    let mut attributes_vec = vec![];
+    while let Tok::AttrBegin = tokens.peek() {
+        let start_loc = tokens.start_loc();
+        tokens.advance()?;
+        let attributes_ = parse_comma_list_after_start(
```

Comment:
I'm curious: is there a particular reason why you want to combine the "#" and "[" in a single AttrBegin token instead of using a single "#" token with a call here to `parse_comma_list` with an LBracket token for the `start_token`? I think it works either way, but maybe I'm missing something.

Reply:
No real reason, I can and will change this